image-f2fs.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022 Tomas Mudrunka <harviecz@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include <confuse.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "genimage.h"
+
+static int f2fs_generate(struct image *image)
+{
+	int ret;
+
+	char *extraargs = cfg_getstr(image->imagesec, "extraargs");
+	char *label = cfg_getstr(image->imagesec, "label");
+
+	extraargs = cfg_getstr(image->imagesec, "extraargs");
+
+	ret = systemp(image, "%s if=/dev/zero of=\"%s\" seek=%lld count=0 bs=1 2>/dev/null",
+		get_opt("dd"), imageoutfile(image), image->size);
+
+	if(ret) return ret;
+
+	ret = systemp(image, "%s %s '%s' %s '%s' > /tmp/fout.txt",
```

Comment:
This looks wrong. If no label is specified then we get an empty argument. Take a look at how `ext2_generate_mke2fs()` handles this.